snapshot.py

```diff
@@ -81,34 +85,47 @@ def read_stats(args):
        (args.host != '' and args.user != 'root'):
         LOGGER.warning("If not running as root you may not see all info.")
 
-    optional_port = ''
-    if args.ssh_port:
-        optional_port = '-p %d' % args.ssh_port
-
+    cmd = "bash -c \"%s\"" % (cmd % (pids, pids))
     if args.host == '':
         LOGGER.info('Loading local procfs files')
-        cmd = "bash -c \"%s\"" % (cmd % (pids, pids))
+        p = Popen(cmd, shell=True, bufsize=-1, stdout=PIPE, stderr=PIPE)
+        stdout = p.stdout
+        stderr = p.stderr
     elif args.host != '':
-        ssh = (
-            "ssh %s %s@%s"
-            " -o UserKnownHostsFile=/dev/null"
-            " -o StrictHostKeyChecking=no"
-            " -o LogLevel=error"
-            % (optional_port, args.user, args.host)
-        )
+        ssh = paramiko.SSHClient()
+        # INSECURE, but reasonable for STBs
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
```

Comment:
I would assume that it's less secure than RejectPolicy (the default). The comment isn't really useful, I'll remove it.